OpenRA.Mods.Common/Activities/Air/HeliReturnToBase.cs

```diff
@@ -46,13 +47,27 @@ public override Activity Tick(Actor self)
 									.Where(a => a.Owner == self.Owner && rearmBuildings.Contains(a.Info.Name))
 									.ClosestTo(self);
 
-				if (nearestHpad == null)
+				if (nearestHpad == null) {
 					return ActivityUtils.SequenceActivities(new Turn(self, initialFacing), new HeliLand(self, true), NextActivity);
-				else
-					return ActivityUtils.SequenceActivities(new HeliFly(self, Target.FromActor(nearestHpad)));
+				} else {
+					var distanceFromHelipad = (nearestHpad.CenterPosition - self.CenterPosition).HorizontalLength;
+
+					//// if no pad is avaliable, move near one and wait
+					if (distanceFromHelipad > 6000) {
```

Comment:
I think making properties accessable is always a good idea, you never know which behaviour a modder wants...